RecoParticleFlow/PFClusterProducer/plugins/SimMappers/RealisticHitToClusterAssociator.h

```diff
@@ -0,0 +1,288 @@
+#ifndef __RecoParticleFlow_PFClusterProducer_RealisticHitToClusterAssociator_H__
+#define __RecoParticleFlow_PFClusterProducer_RealisticHitToClusterAssociator_H__
+/////////////////////////
+// Author: Felice Pantaleo
+// Date:   30/06/2017
+// Email: felice@cern.ch
+/////////////////////////
+#include <vector>
+#include "RealisticCluster.h"
+
+namespace
+{
+//TODO: get number of layers per subdet from geometry
+float getDecayLength(unsigned int layer)
+{
+    if (layer <= 28)
+        return 2.f;
+    if (layer > 28 && layer <= 40)
+        return 1.5f;
+    if (layer > 40)
+        return 1.f;
+
+    return 0.f;
+}
+}
+
+class RealisticHitToClusterAssociator
+{
+        using Hit3DPosition = std::array<float,3>;
+    public:
+        void init(std::size_t numberOfHits, std::size_t numberOfSimClusters,
+                std::size_t numberOfLayers)
+        {
+            hitPosition_.resize(numberOfHits);
```

Comment:
Maybe you should make a struct with all these properties instead of having a bunch of vectors with the same length (so there could be three types, structs with length `numberOfHits`, with length `numberOfSimClusters`, and with length `numberOfLayers`)

(then you could turn some of the old-style loops in other functions into range-based loops)